common/setups/rasr/gmm_system.py

```diff
@@ -851,11 +895,17 @@ def run_monophone_step(self, step_args):
                 name = step_args.training_args["name"]
                 feature_scorer = (trn_c, f"train_{name}")
 
+                recog_args = copy.deepcopy(step_args.recognition_args)
+                if step_args.test_recognition_args is None:
+                    break
+                recog_args.update(step_args.test_recognition_args)
```

Comment:
Maybe we do want to use the same recognition args for test and dev corpora?
```suggestion
                if step_args.test_recognition_args is not None:
                    recog_args.update(step_args.test_recognition_args)
```
or would you rather that
`test_recognition_args = None` -> no test
`test_recognition_args = {}` -> test with same config

Reply:
yes. I want to have no decoding for the `None` case.
Idea is to first evaluate on the `dev` set. Find a good set of parameters and then pick/update the recog args accordingly